yarn-project/simulator/src/avm/journal/trace.ts

```diff
@@ -1,5 +1,24 @@
 import { type UnencryptedL2Log } from '@aztec/circuit-types';
-import { type EthAddress, L2ToL1Message, LogHash } from '@aztec/circuits.js';
+import {
+  type EthAddress,
+  L2ToL1Message,
+  LogHash,
+  MAX_NEW_L2_TO_L1_MSGS_PER_CALL,
+  MAX_NEW_L2_TO_L1_MSGS_PER_TX,
```

Comment:
One reason to consider not doing this in the journal or even in the simulator (emphasis in _consider_): if we are going to throw an error, it should be in the spec.

I guess we do want to fail early in these cases and revert? Somehow it feels different than failing when there's already a nullifier and you try to double-spend. In this case the constraint is imposed by kernel (limitations), while on the nullifiers case it makes sense for the simulator as an autonomous unit.